dbms/src/TableFunctions/TableFunctionRandom.cpp

```diff
@@ -0,0 +1,477 @@
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                UInt64 a = static_cast<UInt64>(generator()) << 32 | static_cast<UInt64>(generator());
+                data[i] = static_cast<UInt64>(a);
+            }
+            break;
+        }
+        case TypeIndex::UInt128:
+            throw Exception("There is no DataType 'UInt128' support.", ErrorCodes::NOT_IMPLEMENTED);
+        case TypeIndex::Int8:
+        {
+            auto & data = typeid_cast<ColumnVector<Int8> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                data[i] = static_cast<Int8>(generator());
+            }
+            break;
+        }
+        case TypeIndex::Int16:
+        {
+            auto & data = typeid_cast<ColumnVector<Int16> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                data[i] = static_cast<Int16>(generator());
+            }
+            break;
+        }
+        case TypeIndex::Int32:
+        {
+            auto & data = typeid_cast<ColumnVector<Int32> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                data[i] = static_cast<Int32>(generator());
+            }
+            break;
+        }
+        case TypeIndex::Int64:
+        {
+            auto & data = typeid_cast<ColumnVector<Int64> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                Int64 a = static_cast<Int64>(generator()) << 32 | static_cast<Int64>(generator());
+                data[i] = static_cast<Int64>(a);
+            }
+            break;
+        }
+        case TypeIndex::Int128:
+            throw Exception("There is no DataType 'Int128' support.", ErrorCodes::NOT_IMPLEMENTED);
+        case TypeIndex::Float32:
+        {
+            auto & data = typeid_cast<ColumnVector<Float32> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            double d = 1.0;
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                d = std::numeric_limits<float>::max();
+                data[i] = (d / pcg32::max()) * generator();
+            }
+            break;
+        }
+        case TypeIndex::Float64:
+        {
+            auto & data = typeid_cast<ColumnVector<Float64> &>(column).getData();
+            data.resize(limit);
+            pcg32 generator(random_seed);
+            double d = 1.0;
+            for (UInt64 i = 0; i < limit; ++i)
+            {
+                d = std::numeric_limits<double>::max();
+                data[i] = (d / pcg32::max()) * generator();
```

Comment:
If we intend to use this table function for fuzz testing, denormals should also be possible.
(but I'm still unaware of the purpose of this function)